google-cloud-trace/lib/google/cloud/trace/faraday_middleware.rb

```diff
@@ -43,17 +43,34 @@ def add_request_labels span, env
           label_keys = Google::Cloud::Trace::LabelKey
 
           set_label labels, label_keys::HTTP_METHOD, env.method
-          set_label labels, label_keys::HTTP_USER_AGENT,
-                    env.request_headers[:user_agent]
           set_label labels, label_keys::HTTP_URL, env.url.to_s
+
+          # Only sets request size if request is not sent yet.
+          unless env.status
+            request_body = env.body || ""
+            set_label labels, label_keys::RPC_REQUEST_SIZE,
+                      request_body.bytesize.to_s
+          end
         end
 
         ##
         # @private Set Trace span labels from response
-        def add_response_labels span, response
-          set_label span.labels,
-                    Google::Cloud::Trace::LabelKey::HTTP_STATUS_CODE,
-                    response.status.to_s
+        def add_response_labels span, env
+          labels = span.labels
+          label_keys = Google::Cloud::Trace::LabelKey
+
+          response = env.response
+          response_body = env.body || ""
```

Comment:
It's not defined as a public method on `Faraday::Env`, but as part of initialization parameter. [Here's the code](https://github.com/lostisland/faraday/blob/4206d48f54ff78a819581f5502e505b53b5c262e/lib/faraday/options.rb#L266).

Now you mentioned, this field seems to be accessible from both places, `env.body` and `env.response.body`. We can use either one here, since the `#add_request_labels` method is accessing the body through `env.body`, and `env.response.body` also makes sense in this method.